azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -934,7 +934,9 @@ public void cancelExecutingJobs(ExecutableFlow exFlow, String userId,
   @Override
   public String submitExecutableFlow(ExecutableFlow exflow, String userId)
     throws ExecutorManagerException {
-    synchronized (exflow) {
+
+    String exFlowKey = exflow.getProjectName() + "." + exflow.getId() + ".submitFlow";
+    synchronized (exFlowKey.intern()) {
```

Comment:
Is this method called from the TriggerScannerThread? 
If so, will it block other flows from being launched?

Also, the concurrent options need to be taken into consideration. e.g. if user specified skip option, the second flow should be skipped.